src/proxy/inbound.rs

```diff
@@ -62,7 +62,10 @@ impl Inbound {
             .map_err(|e| Error::Bind(pi.cfg.inbound_addr, e))?;
         let transparent = super::maybe_set_transparent(&pi, &listener)?;
         // Override with our explicitly configured setting
-        let enable_orig_src = pi.cfg.enable_original_source.unwrap_or(transparent);
+        let enable_orig_src = pi
+            .cfg
+            .explicitly_configure_original_source
+            .unwrap_or(transparent);
```

Comment:
if maybe set_transparent returns an error then we never get here though. we return early at the `?`. Aside from that if maybe_set_returns an error then we are in conflict and we just treat error as meaning false so as to not return early we wind up at a different problem. The active state would be false even though the requested state is true. Setting the variable that contains the active state to be true because the user asked for it means that variable no longer actually reflects the active state so that seems wrong to me.

Reply:
I think we should just `panic` if the config flag is set and we can't do `maybe_set_transparent` because of lack of perms, then we can just use `transparent` in all cases.

Reply:
> if maybe set_transparent returns an error then we never get here though. we return early at the ?

that's true, but it will not always return an error, whether it does or not is dependent on this flag.

Reply:
it either has an error or it contains the state that resulted when we tried to set transparent, yeah? if it's an error then `enable_orig_src` doesn't matter because we never reach it. If its not an error then we really just want what is in transparent

Reply:
> If its not an error then we really just want what is in transparent

unless enable_orig_src was explicitly set to `false` as user intent, in which case we want to discard the result of the attempt and use the user intent.

It's goofy. I think we should use 2 config fields.

Reply:
but if the user asked for false we just don't attempt to set transparent and return false

https://github.com/howardjohn/ztunnel/blob/2eae711bc42832d928f2d16d3dfc7687bd34c09a/src/proxy.rs#L409

Reply:
Intent (and, afaik, implementation) is:
* default, None: try to set transparent (requires privs), if it doesn't work proceed with original_src=false
* Some(true): set transparent or fail. use original src
* Some(false): don't bother trying transparent, do not use original src

The higher level intent is to make it so we gracefully degrade without privs, but can explicitly require/not